src/database/UIDatabase.js

```diff
@@ -6,6 +9,32 @@ export class UIDatabase {
     this.database = database;
   }
 
+  /**
+   * Closes the database, exports the .realm file to android 'Download/mSupplyMobile data'
+   * The app will cease to function (crash) if anything tries to access the database while
+   * it is closed.
+   *
+   * This method should perhaps be added to react-native-databse
+   */
+  exportData(filename = 'msupply-mobile-data') {
+    const realm = this.database.realm; // TODO: refactor away from 'database.database.realm'
+    const realmPath = realm.path;
+    const exportFolder = `${RNFS.ExternalStorageDirectoryPath}/Download/mSupplyMobile_data`;
+
+    const date = new Date();
+    const dateString = `${date.getFullYear()}-${date.getMonth()}-${date.getDay()}T${date.getHours()}-${date.getMinutes()}-${date.getSeconds()}`;
+    const copyFileName = `${filename} ${dateString}`;
+
+    // If the database is not closed, there is a small chance of corrupting the data
+    // if it's currently in a transaction
+    realm.close();
+    RNFS.mkdir(exportFolder)
```

Comment:
would making export dir fail if it already exists ?

Reply:
Nah

>Create a directory at filepath. Automatically creates parents and does not throw if already exists (works like Linux mkdir -p).

https://github.com/itinance/react-native-fs#mkdirfilepath-string-options-mkdiroptions-promisevoid